tests/unit/parts/test_lifecycle.py

```diff
@@ -1811,6 +1811,82 @@ def test_patch_elf(snapcraft_yaml, mocker, new_dir):
     ]
 
 
+def test_patch_elf_with_override_prime(snapcraft_yaml, mocker, new_dir, emitter):
+    """Patch binaries with `enable-patchelf`` and override-prime defined"""
+    mocker.patch("snapcraft.elf._patcher.Patcher._run_patchelf")
+    shutil.copy("/bin/true", "elf.bin")
+    callbacks.register_post_step(parts_lifecycle._patch_elf, step_list=[Step.PRIME])
+
+    mocker.patch(
+        "snapcraft.elf.elf_utils.get_dynamic_linker",
+        return_value="/snap/core22/current/lib64/ld-linux-x86-64.so.2",
+    )
+    mocker.patch(
+        "snapcraft.elf._patcher.Patcher.get_proposed_rpath",
+        return_value=["/snap/core22/current/lib/x86_64-linux-gnu"],
+    )
+    mocker.patch("snapcraft.elf._patcher.Patcher.get_current_rpath", return_value=[])
+
+    # don't load dependencies in a unit test
+    mocker.patch.object(ElfFile, "load_dependencies")
+
+    yaml_data = {
+        "base": "core22",
+        "confinement": "classic",
+        "parts": {
+            "p1": {
+                "plugin": "dump",
+                "source": ".",
+                "build-attributes": ["enable-patchelf"],
+                "override-prime": "craftctl default",
+            }
+        },
+    }
+    project = Project.unmarshal(snapcraft_yaml(**yaml_data))
+
+    parts_lifecycle._run_command(
+        "pack",
+        project=project,
+        parse_info={},
+        assets_dir=Path(),
+        start_time=datetime.now(),
+        parallel_build_count=1,
+        parsed_args=argparse.Namespace(
+            directory=None,
+            output=None,
+            debug=False,
+            destructive_mode=True,
+            shell=False,
+            shell_after=False,
+            use_lxd=False,
+            enable_manifest=False,
+            ua_token=None,
+            parts=["p1"],
+        ),
+    )
+
+    # Simple verification that our expected message is being printed
+    emitter.assert_progress(
+        "Warning: 'enable-patchelf' feature is unsafe if 'override-prime' "
+        "is also enabled. It's not possible to track file changes in the "
+        "prime directory.",
+        permanent=True,
+    )
+
+    # TODO: This is not working as expected, the patchelf is not being called
+    # See https://github.com/canonical/craft-parts/pull/623
+    #
+    # assert run_patchelf_mock.mock_calls == [
```

Comment:
What do you think of something like `run_patchelf_mock.assert_not_called()`?

Reply:
It should call it, but due to the bug it does not. I am not sure if we want to check it is not called.

Reply:
I posit that we should test for 3 reasons:
1. even if the behavior is undesired, the unit test should still test the behavior.  (the same principle behind `xfail(strict=True)`)
2. when craft-parts is fixed, this test will start failing and alert us that the fix was effective
3. commented out code is a code smell